src/intervals/interval_operations/bisect.jl

```diff
@@ -1,29 +1,43 @@
 """
-    bisect(x::BareInterval, α::Real=0.49609375)
-    bisect(x::Interval, α::Real=0.49609375)
+    bisect(x, α=0.5)
+    bisect(x, i, α=0.5)
 
-Split the interval `x` at position `α`, where `α = 0.5` corresponds to the
-midpoint.
+Split an interval `x` at a relative position `α`, where `α = 0.5` corresponds
+to the midpoint.
+
+Split the `i`-th component of a vector `x` at a relative position `α`, where
+`α = 0.5` corresponds to the midpoint.
 """
-function bisect(x::BareInterval{T}, α::Real=0.49609375) where {T<:NumTypes}
-    0 ≤ α ≤ 1 || return throw(DomainError(α, "bisect only accepts a relative position α between 0 and 1."))
+function bisect(x::BareInterval{T}, α::Real=0.5) where {T<:NumTypes}
+    0 ≤ α ≤ 1 || return throw(DomainError(α, "bisect only accepts a relative position α between 0 and 1"))
+    isatomic(x) && return (x, emptyinterval(BareInterval{T}))
     m = scaled_mid(x, α)
     return (_unsafe_bareinterval(T, inf(x), m), _unsafe_bareinterval(T, m, sup(x)))
 end
 
-function bisect(x::Interval, α::Real=0.49609375)
+function bisect(x::Interval, α::Real=0.5)
     bx = bareinterval(x)
-    r1, r2 = bisect(bx, α)
-    d1, d2 = min(decoration(x), decoration(r1), trv), min(decoration(x), decoration(r2), trv)
+    r₁, r₂ = bisect(bx, α)
+    d₁, d₂ = min(decoration(x), decoration(r₁), trv), min(decoration(x), decoration(r₂), trv)
     t = isguaranteed(x)
-    return (_unsafe_interval(r1, d1, t), _unsafe_interval(r2, d2, t))
+    return (_unsafe_interval(r₁, d₁, t), _unsafe_interval(r₂, d₂, t))
+end
+
+function bisect(x::AbstractVector, i::Integer, α::Real=0.5)
```

Comment:
Don't we need to specify that the elements of `x` are `Interval`s or `BareInterval`s?

Reply:
No not really, this code will fail properly if that is not the case.